docs/ref/models/constraints.txt

```diff
@@ -119,3 +119,57 @@ ensures that each user only has one draft.
 
 These conditions have the same database restrictions as
 :attr:`Index.condition`.
+
+``defer``
+---------
+
+.. attribute:: UniqueConstraint.defer
+
+.. versionadded:: 3.1
+
+This controls whether the constraint can be deferred. By default constraints
+cannot be deferred. Deferrable constraints can be created by passing
+``DEFERRABLE_INITIALLY_DEFERRED`` or ``DEFERRABLE_INITIALLY_IMMEDIATE``::
+
+    from django.db.models import UniqueConstraint
+    from django.db.models.constraints import DEFERRABLE_INITIALLY_DEFERRED
+
+    UniqueConstraint(
+        name='unique_name',
+        fields=['name'],
+        defer=DEFERRABLE_INITIALLY_DEFERRED,
+    )
+
+A deferred constraint will not be enforced until the end of the transaction.
+An immediate constraint will be enforced at the time of each write.
+The behavior of a deferrable constraint can be changed until the end of the
+current transaction using ``set_constraints``.
```

Comment:
It should be noted that this will incur a performance hit.

[From PostgreSQL documentation](https://www.postgresql.org/docs/12/sql-createtable.html#id-1.9.3.85.9.4)
> To obtain standard-compliant behavior, declare the constraint as DEFERRABLE but not deferred (i.e., INITIALLY IMMEDIATE). Be aware that this can be significantly slower than immediate uniqueness checking.